pkg/cmd/admin/groups/new.go

```diff
@@ -49,10 +52,14 @@ func NewCmdNewGroup(name, fullName string, f *clientcmd.Factory, out io.Writer)
 				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
 			}
 
-			kcmdutil.CheckErr(options.AddGroup())
+			mapper, _ := f.Object()
+			shortOutput := kcmdutil.GetFlagString(cmd, "output") == "name"
+			kcmdutil.CheckErr(options.AddGroup(mapper, shortOutput))
 		},
 	}
 
+	kcmdutil.AddOutputFlagsForMutation(cmd)
```

Comment:
> @deads2k what benefits from that change do I use to justify that PR?

Currently, the printer and output methods on the `Factory` are very difficult to use during delegation flows.  They rely on having a `cmd` set up with exactly the "right" flags.  We should refactor the `Factory` to specify the actual values they need.

Having the proper object bound in will allow you separate the data from its expression.